phoenix-core/src/main/java/org/apache/phoenix/coprocessor/ChildLinkMetaDataEndpoint.java

```diff
@@ -111,4 +145,162 @@ private PhoenixMetaDataCoprocessorHost getCoprocessorHost() {
+                ServerUtil.throwIOException(sysCatHTable.getName().toString(), t);
+            }
+            // if row found, repair and verifyRowAndRemoveEmptyColumn
+            if (result != null && !result.isEmpty()) {
+                markChildLinkVerified(rowKey, ts, region);
+                scanner.close();
+                childLinkScan.withStartRow(rowKey, true);
+                scanner = region.getScanner(childLinkScan);
+                hasMore = true;
+            }
+            // if not, delete if old enough, otherwise ignore
+            else {
+                deleteIfAgedEnough(rowKey, ts, region);
+                if (restartScanDueToPageFilterRemoval) {
+                    scanner.close();
+                    childLinkScan.withStartRow(rowKey, true);
+                    scanner = region.getScanner(childLinkScan);
+                    hasMore = true;
+                    restartScanDueToPageFilterRemoval = false;
+                }
+            }
+            row.clear();
+        }
+
+        /*
+        Construct row key for SYSTEM.CATALOG from a given SYSTEM.CHILD_LINK row key
+        SYSTEM.CATALOG -> (CHILD_TENANT_ID, CHILD_SCHEMA, CHILD_TABLE, PARENT_TENANT_ID, PARENT_FULL_NAME)
+        SYSTEM.CHILD_LINK -> (PARENT_TENANT_ID, PARENT_SCHEMA, PARENT_TABLE, CHILD_TENANT_ID, CHILD_FULL_NAME)
+         */
+        private byte[] getSysCatRowKey(byte[] childLinkRowKey) {
+            String NULL_DELIMITER = "\0";
+            String[] childLinkRowKeyCols = new String(childLinkRowKey, StandardCharsets.UTF_8).split(NULL_DELIMITER);
+            checkArgument(childLinkRowKeyCols.length == 5);
+            String parentTenantId = childLinkRowKeyCols[0];
+            String parentSchema = childLinkRowKeyCols[1];
+            String parentTable = childLinkRowKeyCols[2];
+            String childTenantId = childLinkRowKeyCols[3];
+            String childFullName = childLinkRowKeyCols[4];
+
+            String parentFullName = SchemaUtil.getTableName(parentSchema, parentTable);
+            String childSchema = SchemaUtil.getSchemaNameFromFullName(childFullName);
+            String childTable = SchemaUtil.getTableNameFromFullName(childFullName);
+
+            String[] sysCatRowKeyCols = new String[] {childTenantId, childSchema, childTable, parentTenantId, parentFullName};
+            return String.join(NULL_DELIMITER, sysCatRowKeyCols).getBytes(StandardCharsets.UTF_8);
+        }
+
+
+        private void deleteIfAgedEnough(byte[] rowKey, long ts, Region region) throws IOException {
+            if ((EnvironmentEdgeManager.currentTimeMillis() - ts) > ageThreshold) {
+                Delete del = new Delete(rowKey);
+                Mutation[] mutations = new Mutation[]{del};
+                region.batchMutate(mutations);
+            }
+        }
+
+
+        private void markChildLinkVerified(byte[] rowKey, long ts, Region region) throws IOException {
+            Put put = new Put(rowKey);
```

Comment:
Will this not create a new ts for the row

Reply:
the timestamp is extracted from the first cell in the row i.e. `long ts = row.get(0).getTimestamp();`

Reply:
I think you may need to use the Put(byte[] row, long ts) constructor.